### PR TITLE
Pipeline maintenance [Part 2]

### PR DESCRIPTION
**Pull request by NoahNsimbe**
**_WHAT DOES THIS PR DO?_**

- [x] Disables weather data ETL tasks
- [x] Adds more validations on data sent to the data warehouse
- [x] Updates the Readme with a description of the schemas used
- [x] Fixes typos on time fields

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
[PLAT-1040](https://airqoteam.atlassian.net/browse/PLAT-1040)

**_HOW DO I TEST OUT THIS PR?_**
Follow instructions in the [README file](https://github.com/airqo-platform/AirQo-api/tree/post-airflow-fixes/src/airflow#running-using-docker)

**_ARE THERE ANY RELATED PRs?_**
#755

### CODE REVIEW

File: src/airflow/README.md [hunk 1]
```diff
@@ -119,5 +119,14 @@ sh clean.sh
 
 ## Working with dags
 
-ETL dags for historical data require you to specify `startDateTime` and `endDateTime` in the dag config using the 
-format `YYYY-MM-ddTHH:mm:ssZ`
+ETL dags for historical data require you to specify `startDateTime` and `endDateTime` in the dag config using the format `YYYY-MM-ddTHH:mm:ssZ`
+
+## BigQuery Schemas
+
```
Comment (NoahNsimbe) [inline src/airflow/README.md:125]: Description for BigQuery table schemas

File: src/airflow/airqo_etl_utils/airqo_utils.py [hunk 2]
```diff
@@ -689,7 +689,7 @@ def restructure_airqo_data_for_bigquery(data: list) -> list:
     for _, data_row in data_df.iterrows():
         device_data = dict(
             {
-                "time": str_to_date(data_row["time"]),
+                "timestamp": str_to_date(data_row["time"]),
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/airqo_utils.py:692]: fixing typos on the time field

File: src/airflow/airqo_etl_utils/schema/measurements.json [hunk 3]
```diff
@@ -9,6 +9,12 @@
         "type": "TIMESTAMP",
         "mode": "REQUIRED"
     },
+    {
+        "name": "time",
+        "type": "TIMESTAMP",
+        "mode": "NULLABLE",
+        "description": "This field is depreciated. Use timestamp"
+    },
```
Comment (NoahNsimbe) [inline src/airflow/airqo_etl_utils/schema/measurements.json:17]: Added missing time field

File: src/airflow/dags/airqo_measurements.py [hunk 4]
```diff
@@ -147,12 +150,14 @@ def load(airqo_data: dict, **kwargs):
...
-    merged_data = merge_data(
-        averaged_airqo_data=data_with_site_ids, weather_data=extracted_weather_data
-    )
-    calibrated_data = calibrate(merged_data)
+    # Temporarily disabling weather data
+    # extracted_weather_data = extract_hourly_weather_data()
+    # merged_data = merge_data(
+    #     averaged_airqo_data=data_with_site_ids, weather_data=extracted_weather_data
+    # )
+    # calibrated_data = calibrate(merged_data)
```
Comment (NoahNsimbe) [inline src/airflow/dags/airqo_measurements.py:158]: disabled weather data ETL task

File: src/airflow/main.py [hunk 5]
```diff
@@ -19,23 +19,40 @@ def kcca_hourly_measurements(start_date_time: str, end_date_time: str):
...
-        path_or_buf="kcca_data_for_bigquery.csv", index=False
+    pd.DataFrame(message_broker_data).to_csv(
+        path_or_buf="kcca_message_broker_data.csv", index=False
+    )
+
+    # Big Query
+    bigquery_data = transform_kcca_hourly_data_for_bigquery(data=kcca_unclean_data)
+    bigquery_data_df = pd.DataFrame(bigquery_data)
+    bigquery_api = BigQueryApi()
+    bigquery_data_df = bigquery_api.validate_data(
```
Comment (NoahNsimbe) [inline src/airflow/main.py:49]: validating data sent to bigquery

Comment (NoahNsimbe) [inline src/airflow/main.py:49]: validating data sent to BigQuery

**Review by Baalmart — APPROVED** [review]
LGTM

File: src/airflow/dags/kcca_measurements.py [hunk 6]
```diff
@@ -14,24 +14,18 @@
...
-            frequency = dag_run.conf["frequency"]
-            start_time = dag_run.conf["startTime"]
-            end_time = dag_run.conf["endTime"]
-        except KeyError:
-            frequency = "hourly"
-            start_time = date_to_str_hours(datetime.utcnow() - timedelta(hours=4))
-            end_time = date_to_str_hours(datetime.utcnow())
+        hour_of_day = datetime.utcnow() - timedelta(hours=1)
+        start_date_time = date_to_str_hours(hour_of_day)
+        end_date_time = datetime.strftime(hour_of_day, "%Y-%m-%dT%H:59:59Z")
```
Comment (123MwanjeMike) [inline src/airflow/dags/kcca_measurements.py:25]: Hi @NoahNsimbe. Looks like you're rounding off to the end of the hour. Is it the intended purpose?

**Review by 123MwanjeMike — COMMENTED** [review]
Thanks @NoahNsimbe.
Looks good to me generally, just a minor inquiry

File: src/airflow/dags/kcca_measurements.py (hunk 6 above)
Comment (NoahNsimbe) [reply src/airflow/dags/kcca_measurements.py:25]: yes, It is intended. In an event where the start and end time are not provided, we default to start and end time of the previous hour.

Comment (123MwanjeMike) [reply src/airflow/dags/kcca_measurements.py:25]: Oh, thanks

**Review by 123MwanjeMike — APPROVED** [review]

**Pull request merged**